src/compiler/program.ts

```diff
@@ -333,7 +333,7 @@ namespace ts {
         if (host.resolveModuleNames) {
             resolveModuleNamesWorker = (moduleNames, containingFile) => host.resolveModuleNames(moduleNames, containingFile).map(resolved => {
                 // An older host may have omitted extension, in which case we should infer it from the file extension of resolvedFileName.
-                if (!resolved || resolved.extension) {

```

Comment:
`resolved.extension` was completely backwards before then? That is, it should have been `!resolved.extension` to even be close to correct?